### ISSUE TITLE
In Work records need to show leave types like sick leave ,casual leave 

### ISSUE THREAD

**Issue by RaagulElangovan** [horilla-opensource/horilla#289]
# Feature Request

## Description
When you apply for leave, the work record should display the specific type of leave applied for instead of marking it as "Absent."
If partial leave is applied, the system should display a combination of leave type and work hours (e.g., "Half Day Present - Sick Leave").

## Use Case:
The user requires an accurate representation of leave in the work record, necessitating clear records for both attendance and leave management. They need to track employee attendance and leave precisely.

## Priority
 High

**Comment by horilla-opensource** [comment]
Hi @RaagulElangovan ,
We have tried to add the leave type to the work records, but due to the multiple color combinations, the work record view has some user experience issues. 
One thing we can do is, we can display the leave type on hover over the record.

With Regards,
**Team Horilla**

**Comment by RaagulElangovan** [comment]
Hi Team, 
As you have issue in displaying the different leave type due to UI issue, Can you provide those leave type data in the export option in excel? 

Thanks,
Raagul

**Comment by horilla-opensource** [comment]
Hi @RaagulElangovan ,
You want the labels inside the cell to be specifically showing the leave type name rather than **ABS** ?

With Regards,
**Team Horilla**

**Comment by RaagulElangovan** [comment]
Yes, the goal is to have the labels inside the cell specifically show the leave type name, such as "Sick Leave" or "Casual Leave," rather than just displaying "ABS." This will make it clearer which type of leave has been applied.

**Comment by horilla-opensource** [comment]
Hi @RaagulElangovan ,
Please update the views.py with the new views.py in the link below.
[views.py](https://drive.google.com/file/d/1rzZQuKy94DNTavwE35hAmuzaDiDM3lQ8/view?usp=sharing)

Please let me know if you have any questions.

With Regards,
**Team Horilla**

**Comment by ArulJacklin** [comment]
"When continuous leave is applied, the first day is correctly recorded as the type of leave applied, but subsequent days are marked as ABS (Absent). Example: If, I apply sick leave continuously for three days first day it showing as a sick leave but next two days it showing as ABS in the work record.
![Screenshot 2024-08-30 135434](https://github.com/user-attachments/assets/461dc438-788a-4114-9a7e-841d41bbf524)

**Comment by horilla-opensource** [comment]
Hi @ArulJacklin ,
Please find the updated code in the below link.
Replace the views.py file with the latest file from the link.
[views.py](https://drive.google.com/file/d/1rzZQuKy94DNTavwE35hAmuzaDiDM3lQ8/view?usp=sharing)

Please let us know if you have any questions.

With Regards,
**Team Horilla**

**Comment by RaagulElangovan** [comment]
Hi Team,

In the Excel file, the half-day present and half-day leave are not reflected correctly; instead, it is displaying a full day of leave as taken.
![image (8)](https://github.com/user-attachments/assets/eaa2a60b-9616-4e0a-b216-32670c859704)

**Comment by horilla-opensource** [comment]
Hi @RaagulElangovan ,
It'll be difficult to have all the details to be included in the single cell. It'll affect the overall readability of the export sheet.
If needed we can add that also.

With Regards,
**Team Horilla**

**Comment by Kuttypillai** [comment]
Hi team,
Can you add like the mentioned image
![image (2)](https://github.com/user-attachments/assets/7d0613e1-b9de-4548-8d2c-e332869109ba)

**Comment by Kuttypillai** [comment]
Hi team, 
when can we expect this feature

**Comment by horilla-opensource** [comment]
Hi @horilla-opensource ,
Find the updated views.py with the suggested feature in the below link.

[views.py](https://drive.google.com/file/d/1xYfXk4YL9lHssUYBNZg6s2N-6oeGS99x/view?usp=sharing)

With Regards,
**Team Horilla**

**Comment by ArulJacklin** [comment]
Hi team ,
We are still encountering an inconsistency during the export process, where the leave type name (e.g., 'Sick Leave' or 'Casual Leave') is not consistently displayed in Excel cells. Although some days correctly show the specific leave type, other days continue to display the generic label '**ABS**,' leading to confusion. We need the export feature to reliably display the exact leave type across all cells to clarify the type of leave applied."

![image](https://github.com/user-attachments/assets/250b4321-4416-46f5-bdc2-49ea63b3d219)

**Comment by RaagulElangovan** [comment]
Hi team,
when can we expect this feature

**Comment by horilla-opensource** [comment]
Hi @RaagulElangovan ,

We are not able to reproduce the issue at our end. 

Please contact our developer at safvan@cybrosys.info for further support.

**Comment by horilla-opensource** [comment]
Hi,
We are closing this issue due to inactivity. It seems that there hasn't been any recent activity or discussion, and we believe it may have been resolved or is no longer relevant. 
If you feel this issue is still important and should be addressed, please feel free to reopen it or create a new issue with updated details. 

With Regards,
**Team Horilla**
